tests/trx_generator/trx_generator.cpp

```diff
@@ -5,200 +5,294 @@
 #include <boost/algorithm/string.hpp>
 #include <eosio/chain/chain_id_type.hpp>
 #include <eosio/chain/name.hpp>
-#include <eosio/chain/asset.hpp>
 #include <fc/bitutil.hpp>
 #include <fc/io/raw.hpp>
+#include <regex>
 
+using namespace std;
 using namespace eosio::chain;
 using namespace eosio;
 using namespace appbase;
 namespace bpo=boost::program_options;
 
 namespace eosio::testing {
-   struct action_pair_w_keys {
-      action_pair_w_keys(eosio::chain::action first_action, eosio::chain::action second_action, fc::crypto::private_key first_act_signer, fc::crypto::private_key second_act_signer)
-            : _first_act(first_action), _second_act(second_action), _first_act_priv_key(first_act_signer), _second_act_priv_key(second_act_signer) {}
 
-      eosio::chain::action _first_act;
-      eosio::chain::action _second_act;
-      fc::crypto::private_key _first_act_priv_key;
-      fc::crypto::private_key _second_act_priv_key;
-   };
+   void trx_generator_base::set_transaction_headers(transaction& trx, const block_id_type& last_irr_block_id, const fc::microseconds& expiration, uint32_t delay_sec) {
+      trx.expiration = fc::time_point::now() + expiration;
+      trx.set_reference_block(last_irr_block_id);
+
+      trx.max_net_usage_words = 0;// No limit
+      trx.max_cpu_usage_ms = 0;   // No limit
+      trx.delay_sec = delay_sec;
+   }
 
-   signed_transaction_w_signer create_transfer_trx_w_signer(const action& act, const fc::crypto::private_key& priv_key, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration, const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
+   signed_transaction_w_signer trx_generator_base::create_trx_w_actions_and_signer(std::vector<action>&& acts, const fc::crypto::private_key& priv_key, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration, const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
       signed_transaction trx;
-      trx.actions.push_back(act);
-      trx.context_free_actions.emplace_back(action({}, config::null_account_name, name("nonce"),
-         fc::raw::pack(std::to_string(nonce_prefix) + ":" + std::to_string(++nonce) + ":" +
-         fc::time_point::now().time_since_epoch().count())));
+      set_transaction_headers(trx, last_irr_block_id, trx_expiration);
+      trx.actions = std::move(acts);
+      trx.context_free_actions.emplace_back(vector<permission_level>(), config::null_account_name, name("nonce"),
+         fc::raw::pack(std::to_string(_config._generator_id) + ":" + std::to_string(nonce_prefix) + ":" + std::to_string(++nonce) + ":" +
+         fc::time_point::now().time_since_epoch().count()));
```

Comment:
These lines could be combined to keep the arg on a single line. This same comment is elsewhere. Aesthetic only, feel free to disregard and resolve yourself.

Reply:
Addressed: 72451aa